cache/cache.go

```diff
@@ -245,84 +260,14 @@ func (c *Cache) downloadHTTP(b *ui.Task, checksum string, uri string, cachePath
 	return cachePath, etag, nil
 }
 
-type writer interface {
-	io.Writer
-	io.WriterAt
-}
-
-// An io.WriterAt that updates a UI task's progress with the bytes written.
-type progressWriterAt struct {
-	w    writer
-	task *ui.Task
-}
-
-func (p2 *progressWriterAt) Write(p []byte) (n int, err error) {
-	n, err = p2.w.Write(p)
-	p2.task.Add(n)
-	return n, errors.WithStack(err)
-}
-
-func (p2 *progressWriterAt) WriteAt(p []byte, off int64) (n int, err error) {
-	n, err = p2.w.WriteAt(p, off)
-	p2.task.Add(n)
-	return n, errors.WithStack(err)
-}
-
-// matches: https://github.com/{OWNER}/{REPO}/releases/download/{TAG}/{ASSET}
-var githubRe = regexp.MustCompile(`^https\://github.com/([^/]+)/([^/]+)/releases/download/([^/]+)/([^/]+)$`)
-
-type githubReleaseInfo struct {
-	owner, repo, tag, asset string
-}
-
-func getGitHubReleaseInfo(uri string) (*githubReleaseInfo, bool) {
-	g := &githubReleaseInfo{}
-	m := githubRe.FindStringSubmatch(uri)
-	if len(m) != 5 {
-		return nil, false
-	}
-	var err error
-	if g.owner, err = url.PathUnescape(m[1]); err != nil {
-		return nil, false
-	}
-	if g.repo, err = url.PathUnescape(m[2]); err != nil {
-		return nil, false
-	}
-	if g.tag, err = url.PathUnescape(m[3]); err != nil {
-		return nil, false
-	}
-	if g.asset, err = url.PathUnescape(m[4]); err != nil {
-		return nil, false
-	}
-	return g, true
-}
-
-func downloadGHPrivate(client *github.Client, ghi *githubReleaseInfo, file string) (w *os.File, response *http.Response, err error) {
-	r, err := client.Releases(fmt.Sprintf("%s/%s", ghi.owner, ghi.repo))
-	if err != nil {
-		return nil, nil, errors.WithStack(err)
-	}
-	asset, err := getAssetURL(r, ghi.tag, ghi.asset)
+func (c *Cache) defaultDownloadStrategy(ctx context.Context, url string) (*http.Response, error) {
```

Comment:
I think we could just inject this to the strategy list at app initialisation? Can't see why it needs to be in `Cache`

Reply:
Because we always want this strategy to be included. If it's optional then the Cache won't work at all.

Reply:
Hmm, what do you mean? If we want a cache that does not use http requests as strategy (only a git cache for example), why would that not work?